hivelocity/instancesV2_test.go

```diff
@@ -110,90 +100,125 @@ func standardMocks(m *mocks.API) {
-	require.Error(t, err)
-}
-
-func Test_InstanceMetadata(t *testing.T) {
-	i2, m := newHVInstanceV2(t)
-	node := newNode()
-	ctx := context.Background()
-	standardMocks(m)
-	metaData, err := i2.InstanceMetadata(ctx, node)
-	require.NoError(t, err)
-	require.Equal(t, &cloudprovider.InstanceMetadata{
-		ProviderID: "14730",
-		NodeAddresses: []corev1.NodeAddress{
-			{
-				Type:    corev1.NodeAddressType("ExternalIP"),
-				Address: "66.165.243.74",
-			},
+	m := mocks.NewAPI(t)
+	i2, node, ctx := newTestData(m)
+	tests := []struct {
+		providerId    int
+		wantBool      bool
+		wantErrString string
+	}{
+		{
+			providerId:    mockDeviceId,
+			wantBool:      false,
+			wantErrString: "",
 		},
-		Zone:   "LAX2",
-		Region: "LAX2",
-	}, metaData)
-
-	node.Spec.ProviderID = "9999999"
-	metaData, err = i2.InstanceMetadata(ctx, node)
-	require.Error(t, err)
-	require.Nil(t, metaData)
+		{
+			providerId:    9999999,
+			wantBool:      false,
+			wantErrString: "i2.API.GetBareMetalDeviceIdResource(deviceId) failed: no such device",
+		},
+	}
+	for _, tt := range tests {
+		node.Spec.ProviderID = fmt.Sprintf("hivelocity://%d", tt.providerId)
+		gotBool, gotErr := i2.InstanceShutdown(ctx, node)
+		msg := fmt.Sprintf("Input: providerId=%+v", tt.providerId)
+		if tt.wantErrString == "" {
+			require.NoError(t, gotErr, msg)
+		} else {
+			require.Error(t, gotErr, msg)
+			require.Equal(t, tt.wantErrString, gotErr.Error(), msg)
+		}
+		require.Equal(t, tt.wantBool, gotBool, msg)
+	}
 }
 
-func Test_getInstanceTypeFromTags(t *testing.T) {
-	type args struct {
-		tags     []string
-		deviceId int32
-	}
+func Test_InstanceMetadata(t *testing.T) {
+	m := mocks.NewAPI(t)
+	i2, node, ctx := newTestData(m)
 	tests := []struct {
-		name string
-		args args
-		want string
+		providerId    int
+		wantMetaData  *cloudprovider.InstanceMetadata
+		wantErrString string
 	}{
-		{"empty slice returns empty string", args{[]string{}, 1}, ""},
-		{"invalid label value will be skipped", args{[]string{"instance-type=&"}, 1}, ""},
-		{"valid label value will be used", args{[]string{"instance-type=abc"}, 1}, "abc"},
+		{
+			providerId: mockDeviceId,
+			wantMetaData: &cloudprovider.InstanceMetadata{
+				ProviderID: "14730",
+				NodeAddresses: []corev1.NodeAddress{
+					{
+						Type:    corev1.NodeAddressType("ExternalIP"),
+						Address: "66.165.243.74",
+					},
+				},
+				Zone:         "LAX2",
+				Region:       "LAX2",
+				InstanceType: "bare-metal-x",
+			},
+			wantErrString: "",
+		},
+		{
+			providerId:    9999999,
+			wantMetaData:  nil,
+			wantErrString: "i2.API.GetBareMetalDeviceIdResource(deviceId) failed: no such device",
+		},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			if got := getInstanceTypeFromTags(tt.args.tags, tt.args.deviceId); got != tt.want {
-				t.Errorf("getInstanceTypeFromTags() = %v, want %v", got, tt.want)
-			}
-		})
+		node.Spec.ProviderID = fmt.Sprintf("hivelocity://%d", tt.providerId)
```

Comment:
Yeah, here we shouldn't need to add the prefix. The ProviderID should have it already.